server/routes/induction/common/workedBeforeController.ts

```diff
@@ -0,0 +1,40 @@
+import { NextFunction, Request, RequestHandler, Response } from 'express'
+import type { InductionDto } from 'inductionDto'
+import type { WorkedBeforeForm } from 'inductionForms'
+import InductionController from './inductionController'
+import WorkedBeforeView from './workedBeforeView'
+import YesNoValue from '../../../enums/yesNoValue'
+
+/**
+ * Abstract controller class defining functionality common to both the Create and Update Induction journeys.
+ */
+export default abstract class WorkedBeforeController extends InductionController {
+  constructor() {
+    super()
+  }
+
+  /**
+   * Returns the WorkedBefore view; suitable for use by the Create and Update journeys.
+   */
+  getWorkedBeforeView: RequestHandler = async (req: Request, res: Response, next: NextFunction): Promise<void> => {
+    const { prisonerSummary, inductionDto } = req.session
+
+    const workedBeforeForm = req.session.workedBeforeForm || toWorkedBeforeForm(inductionDto)
+    req.session.workedBeforeForm = undefined
+
+    const view = new WorkedBeforeView(
+      prisonerSummary,
+      this.getBackLinkUrl(req),
+      this.getBackLinkAriaText(req),
+      workedBeforeForm,
+      req.flash('errors'),
+    )
+    return res.render('pages/induction/worked-before/index', { ...view.renderArgs })
```

Comment:
I think Adam has created the view in the folder `pages/induction/hasWorkedBefore` (because that's the exact same path that is in the CIAG UI codebase)

Either this line will need to change, or you'll need to rename the directory that Adam created.

I don't mind too much if you choose to rename, but I do want to be careful and very mindful of where we do deviate from the CIAG UI.

Reply:
FWIW, whilst for the most part I want to keep the CIAG UI naming conventions as much as possible (for now), in this case I think renaming it to worked before is worth doing. If nothing else its a better name 😁 , but it does also give us the scope to add other questions to the page that are related to whether they have worked before or not (ie. it doesnt feel as tightly coupled to one specific question)